### PR TITLE
Move the clearing of indirect arguments to a seperate pass

### PR DESCRIPTION
**Pull request by ashleysmithgpu**
The order of execution of FilterCS() within a Dispatch() is not determinate, so we need to run the clear in a previous pass.

### CODE REVIEW

File: amd_geometryfx/src/AMD_GeometryFX_Filtering.cpp [hunk 1]
```diff
@@ -341,6 +342,11 @@ class SmallBatchChunk
 
         context->CSSetShader (filterShader, nullptr, 0);
 
+        if (currentBatchCount_ > 1)
```
Comment (Anteru) [inline amd_geometryfx/src/AMD_GeometryFX_Filtering.cpp]: What does this bit do? That seems wrong.

**Review by Anteru — CHANGES_REQUESTED** [review]
The .inc files needs to get updated as well (by compiling the shader and re-generating the header.)

**Comment by ashleysmithgpu** [comment]
Updated the shader .inc files as requested, the extra dispatch was an accidental commit from me testing some code, apologies for that.

**Comment by Anteru** [comment]
Thanks a lot for the quick turnaround on the changes!

**Pull request merged**